src/game_inventory.cpp

```diff
@@ -231,6 +231,110 @@ item_location game::inv_for_gunmod( const item &gunmod, const std::string &title
                          title, -1, _( "You don't have any guns to modify." ) );
 }
 
+class read_inventory_preset: public inventory_selector_preset
+{
+    public:
+        read_inventory_preset( const player &p ) : p( p ) {
+            static const std::string unknown( _( "<color_dkgray>?</color>" ) );
+            static const std::string martial_arts( _( "martial arts" ) );
+
+            append_cell( [ this, &p ]( const item_location & loc ) -> std::string {
+                if( loc->type->can_use( "MA_MANUAL" ) ) {
+                    return martial_arts;
+                }
+                if( !is_known( loc ) ) {
+                    return unknown;
+                }
+                const auto &book = get_book( loc );
+                if( book.skill && p.get_skill_level( book.skill ).can_train() ) {
+                    return string_format( _( "%s to %d" ), book.skill->name().c_str(), book.level );
+                }
+                return std::string();
+            }, _( "TRAINS" ), unknown );
+
+            append_cell( [ this ]( const item_location & loc ) -> std::string {
+                if( !is_known( loc ) ) {
+                    return unknown;
+                }
+                const auto &book = get_book( loc );
+                const int unlearned = book.recipes.size() - get_known_recipes( book );
+
+                return unlearned > 0 ? std::to_string( unlearned ) : std::string();
+            }, _( "RECIPES" ), unknown );
+
+            append_cell( [ this ]( const item_location & loc ) -> std::string {
+                if( !is_known( loc ) ) {
+                    return unknown;
+                }
+                const int fun = get_book( loc ).fun;
+                if( fun > 0 ) {
+                    return string_format( "<good>+%d</good>", fun );
+                } else if( fun < 0 ) {
+                    return string_format( "<bad>%d</bad>", fun );
+                }
+                return std::string();
+            }, _( "FUN" ), unknown );
+
+            append_cell( [ this, &p ]( const item_location & loc ) -> std::string {
+                if( !is_known( loc ) ) {
+                    return unknown;
+                }
+                std::vector<std::string> dummy;
+                const player *reader = p.get_book_reader( *loc, dummy );
+                if( reader == nullptr ) {
+                    return std::string();  // Just to make sure
+                }
+
+                const int actual_time = p.time_to_read( *loc, *reader ) / MOVES( 1 );
```

Comment:
A little more documentation could be helpful explaining what these variables actual are